### PR TITLE
fix: do not open browser when open parameter is set to false

### PR DESCRIPTION
**Pull request by esskar**
## Summary

Allows to disable browser opening by setting the open parameter to false.

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/src/server/open.ts [hunk 1]
```diff
@@ -107,11 +107,15 @@ export function resolveUrl(str: string, base: string): string {
...
+): {
+  targets: string[];
+  before?: () => Promise<void> | void;
+  skip?: boolean;
+} => {
   const { open } = config.server;
 
   if (typeof open === 'boolean') {
-    return { targets: [] };
+    return { targets: [], skip: !open };
```
Comment (chenjiahan) [inline packages/core/src/server/open.ts:118]: This doesn't seem necessary, the server plugin already checks if `open` is false:

https://github.com/web-infra-dev/rsbuild/blob/main/packages/core/src/plugins/server.ts#L15

Comment (esskar) [reply packages/core/src/server/open.ts:118]: Then there must be something else wrong as 

```
import { defineConfig } from '@rsbuild/core';
import { pluginReact } from '@rsbuild/plugin-react';

export default defineConfig({
  plugins: [pluginReact()],
  server: {
    open: false,
  },
});
```
still opens the browser after the server started.

Comment (chenjiahan) [reply packages/core/src/server/open.ts:118]: Are you using the `--open` CLI flag? The CLI flag takes precedence over the configurations,

Comment (esskar) [reply packages/core/src/server/open.ts:118]: oh no, it is part of the package dev script. thanks. sorry for the confusion. will close it.

**Pull request closed without merging**